app/src/main/java/org/asteroidos/sync/services/SynchronizationService.java

```diff
@@ -31,198 +33,241 @@
+        if (mDevice != null) {
+            try {
+                replyTo.send(Message.obtain(null, MSG_SET_STATUS, mState, 0));
+            } catch (RemoteException | NullPointerException ignored) {
             }
-        });
+        }
     }
 
-    void handleSetDevice(String macAddress) {
-        SharedPreferences.Editor editor = mPrefs.edit();
-        editor.putString(MainActivity.PREFS_DEFAULT_MAC_ADDR, macAddress);
-
-        if(macAddress.isEmpty()) {
-            if(mState != STATUS_DISCONNECTED) {
-                mScreenshotService.unsync();
-                mWeatherService.unsync();
-                mNotificationService.unsync();
-                mMediaService.unsync();
-                mTimeService.unsync();
-                mDevice.disconnect();
-                mDevice.unbond();
-            }
-            mDevice = null;
-            editor.putString(MainActivity.PREFS_DEFAULT_LOC_NAME, "");
-        } else {
-            mDevice = mBleMngr.getDevice(macAddress);
+    final public void unsyncServices() {
+        bleServices.forEach(IService::unsync);
+        nonBleServices.forEach(IService::unsync);
+    }
 
-            String name = mDevice.getName_normalized();
-            try {
-                Message answer = Message.obtain(null, MSG_SET_LOCAL_NAME);
-                answer.obj = name;
-                replyTo.send(answer);
+    final public void syncServices() {
+        bleServices.forEach(IService::sync);
+        nonBleServices.forEach(IService::sync);
+    }
 
-                replyTo.send(Message.obtain(null, MSG_SET_STATUS, mState, 0));
-            } catch (RemoteException | NullPointerException ignored) {}
+    @Override
+    public final ConnectionState getConnectionState() {
+        ConnectionState state = null;
+        switch (mState) {
+            case (STATUS_CONNECTED):
+                state = ConnectionState.STATUS_CONNECTED;
+                break;
+            case (STATUS_CONNECTING):
+                state = ConnectionState.STATUS_CONNECTING;
+                break;
+            case (STATUS_DISCONNECTED):
+                state = ConnectionState.STATUS_DISCONNECTED;
+                break;
+        }
+        return state;
+    }
+
+    @Override
+    public final void send(UUID characteristic, byte[] data, IConnectivityService service) {
+        mBleMngr.send(characteristic, data);
+        Log.d(TAG, characteristic.toString() + " " + Arrays.toString(data));
+    }
 
-            editor.putString(MainActivity.PREFS_DEFAULT_LOC_NAME, name);
+    @Override
+    public final void registerBleService(IConnectivityService service) {
+        boolean success = bleServices.add(service);
+        Log.d(TAG, "BLE Service registered: " + success + service.getServiceUUID());
+    }
+
+    @Override
+    public final void unregisterBleService(UUID serviceUUID) {
+        for (IConnectivityService service : bleServices) {
+            if (service.getServiceUUID().equals(serviceUUID)) {
```

Comment:
You wrote getServiceByUUID to do precisely that, you might as well use it ;) it will also be more readable